relay-server/src/services/store.rs

```diff
@@ -940,15 +973,229 @@ impl StoreService {
             },
         )?;
 
-        self.outcome_aggregator.send(TrackOutcome {
-            category: DataCategory::SpanIndexed,
-            event_id: None,
-            outcome: Outcome::Accepted,
-            quantity: 1,
-            remote_addr: None,
-            scoping,
-            timestamp: received_at,
-        });
+        Ok(())
+    }
+
+    fn inner_produce_protobuf_span(
+        &self,
+        scoping: Scoping,
+        received_at: DateTime<Utc>,
+        event_id: Option<EventId>,
+        retention_days: u16,
+        span: SpanKafkaMessage,
+    ) -> Result<(), StoreError> {
+        let mut trace_item = TraceItem {
+            item_type: TraceItemType::Span.into(),
+            organization_id: scoping.organization_id.value(),
+            project_id: scoping.project_id.value(),
+            received: Some(Timestamp {
+                seconds: safe_timestamp(received_at) as i64,
+                nanos: 0,
+            }),
+            retention_days: retention_days.into(),
+            timestamp: Some(Timestamp {
+                seconds: span.start_timestamp_precise as i64,
+                nanos: 0,
+            }),
+            trace_id: span.trace_id.to_string(),
+            item_id: u128::from_str_radix(span.span_id, 16)
+                .unwrap_or_default()
+                .to_le_bytes()
+                .to_vec(),
+            attributes: Default::default(),
+            client_sample_rate: 1.0,
+            server_sample_rate: 1.0,
+        };
+
+        if let Some(data) = span.data {
+            for (key, raw_value) in data {
+                let Some(raw_value) = raw_value else { continue };
+                let json_value = serde_json::Value::deserialize(raw_value).unwrap();
```

Comment:
While this shouldn't fail, I think we should just continue on error, maybe with a `relay_log::error`.

```suggestion
                let json_value = match serde_json::Value::deserialize(raw_value) {
                    Ok(value) => value,
                    Err(err) => relay_log::error!(error = &err as &dyn std::error::Error, "failed to deserialize span data value: {key}");
                };
```